File: Petter Bjornstad/Data Harmonization/gomez_functions.py
# Functions for Gomez calculations

data["erpf_raw_plasma_seconds"] = data["erpf_raw_plasma"]/60
data["gfr_raw_plasma_seconds"] = data["gfr_raw_plasma"]/60

# Filtration Fraction
data["ff"] = data["gfr_raw_plasma"]/data["erpf_raw_plasma"] 

# Kfg for group (T1D/T2D kfg: 0.1012, Control kfg: 0.1733)
data["kfg"] = np.select([data["group"].eq("1"), data["group"].eq("2")], [0.1012, 0.1733]) 

# Filtration pressure across glomerular capillaries
data["deltapf"] = (data["gfr_raw_plasma"]/60)/data["kfg"] 

# Plasma protein mean concentration
data["cm"] = (data["bl_tot_protein"]/data["ff"])*np.log(1/(1-data["ff"])) 

# Pi G (Oncotic pressure)
data["pg"] = 5*(data["cm"]-2)

# Glomerular Pressure
data["glomerular_pressure"] = data["pg"] + data["deltapf"] + 10

# Renal Blood Flow
data["rbf"] = (data["erpf_raw_plasma"]) / (1 - data["hct_210"]/100)
data["rbf_seconds"] = (data["erpf_raw_plasma_seconds"]) / (1 - data["hct_210"]/100)

# Renal Vascular Resistance (mmHg*l^-1*min^-1)
data["rvr"] = data["map"] / data["rbf"]

# Efferent Arteriorlar Resistance 
data["re"] = (data["gfr_raw_plasma_seconds"]) / (data["kfg"] * (data["rbf_seconds"] - (data["gfr_raw_plasma_seconds"]))) * 1328

# Afferent Arteriolar Resistance
data["ra"] = ((data["map"] - data["glomerular_pressure"]) / data["rbf_seconds"]) * 1328

File: Petter Bjornstad/Data Harmonization/gomez_functions.R
library(dplyr)

data <- data %>%
  mutate(
    erpf_raw_plasma_seconds = erpf_raw_plasma / 60,
    gfr_raw_plasma_seconds = gfr_raw_plasma / 60,
    
    # Filtration Fraction
    ff = gfr_raw_plasma / erpf_raw_plasma,
    
    # Kfg for group (T1D/T2D: 0.1012, Control: 0.1733)
    kfg = case_when(
      group == "1" ~ 0.1012,
      group == "2" ~ 0.1733,
      TRUE ~ NA_real_
    ),
    
    # Filtration pressure across glomerular capillaries
    deltapf = (gfr_raw_plasma / 60) / kfg,
    
    # Plasma protein mean concentration
    cm = (bl_tot_protein / ff) * log(1 / (1 - ff)),
    
    # Pi G (Oncotic pressure)
    pg = 5 * (cm - 2),
    
    # Glomerular Pressure
    glomerular_pressure = pg + deltapf + 10,
    
    # Renal Blood Flow
    rbf = erpf_raw_plasma / (1 - hct_210 / 100),
    rbf_seconds = erpf_raw_plasma_seconds / (1 - hct_210 / 100),
    
    # Renal Vascular Resistance (mmHg*l^-1*min^-1)
    rvr = map / rbf,
    
    # Efferent Arteriolar Resistance
    re = (gfr_raw_plasma_seconds / (kfg * (rbf_seconds - gfr_raw_plasma_seconds))) * 1328,
    
    # Afferent Arteriolar Resistance
    ra = ((map - glomerular_pressure) / rbf_seconds) * 1328
  )